internal/command/deploy/strategy_bluegreen.go

```diff
@@ -507,18 +507,46 @@ func (bg *blueGreen) DestroyBlueMachines(ctx context.Context) error {
 	ctx, span := tracing.GetTracer().Start(ctx, "destroy_blue_machines")
 	defer span.End()
 
+	const parallelism = 16
+
+	ch := make(chan *machineUpdateEntry, len(bg.blueMachines))
 	for _, gm := range bg.blueMachines {
-		if bg.isAborted() {
-			return ErrAborted
-		}
-		err := gm.leasableMachine.Destroy(ctx, true)
-		if err != nil {
-			bg.hangingBlueMachines = append(bg.hangingBlueMachines, gm.launchInput.ID)
-			continue
-		}
+		ch <- gm
+	}
+	close(ch)
+
+	var mu sync.Mutex
+	var wg sync.WaitGroup
+	for i := 0; i < parallelism; i++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+
+			for gm := range ch {
```

Comment:
it will spawn up to 16 "workers" (goroutines) that consumes from a common queue ( the`ch` channel) and processes messages as soon as the worker is ready to consume them.

This is typical go concurrency handling that not done well have some gotchas (not saying Ben did it wrong, in fact I can't spot anything bad in this code) but I think abstractions like https://github.com/sourcegraph/conc makes the code more readable and less prone to common mistakes done by devs like myself

Reply:
Thanks for the heads up, @dangra. I replaced the implementation with a conc pool here: https://github.com/superfly/flyctl/pull/3335/commits/3502635efc6c7256cb31cffd95e396249d5d4119